source/code-snippets/transactions/txn-convenient.js

```diff
@@ -0,0 +1,83 @@
+import { MongoError, MongoClient, ObjectId } from "mongodb";
+
+const uri =
+  "<connection string uri>";
+const client = new MongoClient(uri);
+await client.connect();
+
+await client
+  .db("testdb")
+  .collection("inventory")
+  .insertMany([
+    // start-inventory
+    { item: "sunblock, 8 oz", qty: 85, price: 6.0 },
+    { item: "beach chair", qty: 30, price: 25.0 }
+    // end-inventory
+  ]);
+
+const order1 = [
+// start-order-successful
+  { item: "sunblock, 8 oz", qty: 3 },
+  { item: "beach chair", qty: 1 }
+// end-order-successful
+];
+
+const order2 = [
+// start-order-fail
+  { item: "volleyball", qty: 1 }
+// end-order-fail
+];
+
+// start-transaction
+const txnResult = await client.withSession(async (session) =>
+  session
+    .withTransaction(async () => {
+      const invColl = client.db("testdb").collection("inventory");
+      const recColl = client.db("testdb").collection("records");
+
+      let total = 0;
+      for (let i = 0; i < order.length; i++) {
```

Comment:
```suggestion
      for (const item of orders) {
```

`order` -> `orders`, but also I'm not seeing it defined, should it be `order1` and `order2`?

Reply:
This code is not necessarily meant to be runnable, but i can add the `orders` variable that the user can set to `order1` or `order2`